scripts/generate-app-config/generateAppConfig.js

```diff
@@ -0,0 +1,98 @@
+const fse = require('fs-extra');
+const path = require('path');
+const writeConfig = require('./writeConfig');
+const generateRouteConfig = require('./generateRouteConfig');
+const generateNameConfig = require('./generateNameConfig');
+const generateExtensions = require('./generateExtensions');
+const generateUtilitiesConfig = require('./generateUtilitiesConfig');
+const generateNavigationItems = require('./generateNavigationItems');
+const generatePagesConfig = require('./generatePagesConfig');
+const ImportAggregator = require('./generation-objects/ImportAggregator');
+
+/**
+* Writes out a file consiting of the config and imports with the given file name to the specified path.
+* Returns an object representing an imported variable.
+*/
+const addConfig = (config, fileName, buildPath, fs, imports) => {
+  if (config) {
+    writeConfig(config, fileName, buildPath, fse);
+    const name = path.parse(fileName).name;
+    return imports.addImport(`./${name}`, name);
+  }
+  return undefined;
+};
+
+/**
+* Writes out a file consiting of the config and imports with the given file name to the specified path.
+* Returns an object representing an imported variable.
+*/
+const generateAppConfig = (siteConfig, production, verbose) => {
+  const imports = new ImportAggregator();
+
+  const { appConfig, navConfig, themeImports } = siteConfig;
+
+  const rootPath = path.join(process.cwd(), 'dev-site-config');
```

Comment:
Does this need to be rootPath, not processPath to correctly generate the app config for package-level tests?
```
  const rootPath = processPath.includes('packages') ? processPath.split('packages')[0] : processPath;
```

Reply:
nope, we build a subset site, just for the package being tested.